drivers/power/lt7170/lt7170.c

```diff
@@ -0,0 +1,1074 @@
+		return lt7170_data2reg_ieee754(dev, data, reg, MILLI);
+	default:
+		return -EINVAL;
+	}
+}
+
+static int lt7170_reg2data(struct lt7170_dev *dev, uint8_t cmd,
+			   uint16_t reg, int *data)
+{
+	switch (cmd) {
+	case LT7170_VOUT_COMMAND:
+	case LT7170_VOUT_MAX:
+	case LT7170_VOUT_MARGIN_HIGH:
+	case LT7170_VOUT_MARGIN_LOW:
+	case LT7170_FREQUENCY_SWITCH:
+	case LT7170_VIN_ON:
+	case LT7170_VIN_OFF:
+	case LT7170_VOUT_OV_FAULT_LIMIT:
+	case LT7170_VOUT_OV_WARN_LIMIT:
+	case LT7170_VOUT_UV_WARN_LIMIT:
+	case LT7170_VOUT_UV_FAULT_LIMIT:
+	case LT7170_VIN_UV_WARN_LIMIT:
+	case LT7170_READ_VIN:
+	case LT7170_READ_VOUT:
+	case LT7170_MFR_READ_EXTVCC:
+	case LT7170_MFR_READ_ITH:
+	case LT7170_MFR_VOUT_PEAK:
+	case LT7170_MFR_VIN_PEAK:
+	case LT7170_IOUT_CAL_OFFSET:
+	case LT7170_IOUT_OC_WARN_LIMIT:
+	case LT7170_OT_FAULT_LIMIT:
+	case LT7170_OT_WARN_LIMIT:
+	case LT7170_READ_IOUT:
+	case LT7170_READ_TEMPERATURE_1:
+	case LT7170_READ_FREQUENCY:
+	case LT7170_MFR_READ_ASEL:
+	case LT7170_MFR_IOUT_PEAK:
+	case LT7170_MFR_RETRY_DELAY:
+	case LT7170_MFR_TEMPERATURE_1_PEAK:
+	case LT7170_MFR_READ_PWM_CFG:
+	case LT7170_MFR_READ_VOUT_CFG:
+	case LT7170_MFR_DISCHARGE_THRESHOLD:
+	case LT7170_MFR_PGOOD_DELAY:
+	case LT7170_MFR_NOT_PGOOD_DELAY:
+		return lt7170_reg2data_ieee754(dev, reg, data, MILLI);
+	default:
+		return -EINVAL;
+	}
+}
+
+/**
+ * @brief Perform packet-error checking via CRC
+ * @param dev - Device structure
+ * @param cmd - PMBus command
+ * @param buf - Pointer to the first packet buffer
+ * @param nbytes - Packet buffer size
+ * @param op - Operation performed. 0 for write, 1 for read
+ * @return PEC code of the PMBus packet
+ */
+static uint8_t lt7170_pec(struct lt7170_dev *dev, uint8_t cmd, uint8_t *buf,
+			  size_t nbytes, uint8_t op)
+{
+	uint8_t crc_buf[nbytes + op + 2];
+
+	crc_buf[0] = (dev->i2c_desc->slave_address << 1);
+	crc_buf[1] = cmd;
+	if (op)
+		crc_buf[2] = (dev->i2c_desc->slave_address << 1) | 1;
+
+	if (buf != NULL && nbytes > 0)
+		memcpy(&crc_buf[2 + op], buf, nbytes);
+
+	return no_os_crc8(lt7170_crc_table, crc_buf, nbytes + op + 2, 0);
+}
+
+/**
+ * @brief Initialize the device structure
+ * @param device - Device structure
+ * @param init_param - Initialization parameters
+ * @return 0 in case of success, negative error code otherwise
+ */
+int lt7170_init(struct lt7170_dev **device,
+		struct lt7170_init_param *init_param)
+{
+	struct lt7170_dev *dev;
+	int ret;
+	uint16_t word;
+	uint8_t *block;
+
+	block = (uint8_t *)no_os_malloc(lt7170_info[init_param->chip_id].name_size);
+	if (!block)
+		return -ENOMEM;
+
+	dev = (struct lt7170_dev *)no_os_calloc(1, sizeof(struct lt7170_dev));
+	if (!dev) {
+		ret = -ENOMEM;
+		goto free_block;
+	}
+
+	/* Initialize I2C */
+	ret = no_os_i2c_init(&dev->i2c_desc, init_param->i2c_init);
+	if (ret)
+		goto i2c_err;
+
+	/* Identify device */
+	ret = lt7170_read_word(dev, LT7170_MFR_SPECIAL_ID, &word);
+	if (word != LT7170_SPECIAL_ID_VALUE) {
+		ret = -EIO;
+		goto dev_err;
+	}
```

Comment:
Do we need to check this? If the MCU can't communicate with the device, the next check would fail anyway, right?